src/test/java/com/dangol/dangolsonnimbackend/boss/service/impl/BossServiceImplTest.java

```diff
@@ -1,11 +1,78 @@
 package com.dangol.dangolsonnimbackend.boss.service.impl;
 
+import com.dangol.dangolsonnimbackend.boss.domain.Boss;
+import com.dangol.dangolsonnimbackend.boss.dto.BossSignupRequestDTO;
+import com.dangol.dangolsonnimbackend.boss.repository.BossRepository;
+import com.dangol.dangolsonnimbackend.boss.repository.dsl.BossQueryRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.crypto.password.PasswordEncoder;
 
 import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
-class BossServiceImplTest {
+public class BossServiceImplTest {
 
+    @Mock
+    private BossRepository bossRepository;
+    @Mock
+    private BossQueryRepository bossQueryRepository;
+    @Mock
+    private PasswordEncoder passwordEncoder;
+
+    @InjectMocks
+    private BossServiceImpl bossServiceImpl;
+
+    private BossSignupRequestDTO dto;
+
+    @BeforeEach
+    void setUp() {
+        dto = new BossSignupRequestDTO();
+        dto.setName("Test");
+        dto.setPassword("password");
+        dto.setEmail("test@test.com");
+        dto.setBossPhoneNumber("01012345678");
+        dto.setStoreRegisterNumber("1234567890");
+        dto.setStoreRegisterName("Test Store");
+        dto.setMarketingAgreement(true);
+    }
+
+    @Test
+    void givenExistingStoreRegisterName_whenSignup_thenThrowException() {
+        when(bossQueryRepository.existsBySrn(anyString())).thenReturn(true);
+        assertThrows(RuntimeException.class, () -> bossServiceImpl.signup(dto));
+    }
+
+    @Test
+    void givenExistingBossPhoneNumber_whenSignup_thenThrowException() {
+        when(bossQueryRepository.existsBySrn(anyString())).thenReturn(false);
+        when(bossQueryRepository.existsByBpn(anyString())).thenReturn(true);
+        assertThrows(RuntimeException.class, () -> bossServiceImpl.signup(dto));
+    }
+
+    @Test
+    void givenExistingEmail_whenSignup_thenThrowException() {
+        when(bossQueryRepository.existsBySrn(anyString())).thenReturn(false);
+        when(bossQueryRepository.existsByBpn(anyString())).thenReturn(false);
+        when(bossQueryRepository.existsByEmail(anyString())).thenReturn(true);
+        assertThrows(RuntimeException.class, () -> bossServiceImpl.signup(dto));
+    }
+
+    @Test
+    void givenNotExistingSrnBpnEmail_whenSignup_thenSuccess() {
```

Comment:
메소드명이 너무 길어져서 축약해야 한다면 주석을 달아서 어떤 테스트를 진행하는지, 어떤 메소드명을 가리키는지 적어놓는 것도 팀원들에게 좋을 것 같습니다! 축약어를 찾아보고 이해하는 것에도 시간이 소요될 수 있을 것 같아요
예시) // 사업자번호, 사장님 번호, 이메일 모두 중복이 아닐시 ...

Reply:
https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea#:~:text=Given_Preconditions_When_StateUnderTest_Then_ExpectedBehavior

진욱님과 대화를하고 BDD 패턴의 기반한 네이밍 규칙을 따르기로 했습니다
해당 네이밍은 조금 길어 질 수 있는데 혹시 추천할만한 네이밍 컨벤션이 있을까요?

Reply:
넵, 네이밍 규칙은 이해했으며 중간 부분에 있는 "SrnBpn" 부분에 대한 피드백으로 주석을 달아놓는 것은 어떤지 말씀드린 거라서 예시로 추가해 놓겠습니다!
```suggestion
    void givenNotExistingSrnBpnEmail_whenSignup_thenSuccess() {
    // 사업자번호, 사장님 번호, 이메일 모두 중복이 아닐시 로그인 성공 테스트입니다.
```

Reply:
아하 ㅎㅎ 제가 리뷰를 잘못 이해했었네요!! 길어진다면 주석을 달아놓아 이해하기 편하도록 작성하도록 하겠습니다.